src/plugins/telemetry_collection_manager/server/plugin.ts

```diff
@@ -227,15 +227,21 @@ export class TelemetryCollectionManagerPlugin
       if (statsCollectionConfig) {
         try {
           const usageData = await this.getUsageForCollection(collection, statsCollectionConfig);
-          if (usageData.length) {
+          const optedInUsageData = usageData.filter(isClusterOptedIn);
+          if (optedInUsageData.length) {
             this.logger.debug(`Got Usage using ${collection.title} collection.`);
             if (config.unencrypted) {
-              return usageData;
+              return optedInUsageData;
             }
 
-            return encryptTelemetry(usageData.filter(isClusterOptedIn), {
```

Comment:
This is the _main_ culprit we didn't notice about this issue earlier. The filter was only applied to the encrypted form so the `unencrypted=true` requests showed all the entries.